agent/consul/state/catalog.go

```diff
@@ -1869,6 +1876,16 @@ func (s *Store) deleteCheckTxn(tx *memdb.Txn, idx uint64, node string, checkID t
 			if err = tx.Insert("index", &IndexEntry{serviceIndexName(existing.ServiceName), idx}); err != nil {
 				return fmt.Errorf("failed updating index: %s", err)
 			}
+
+			svcRaw, err := tx.First("services", "id", existing.Node, existing.ServiceID)
+			if err != nil {
+				return fmt.Errorf("failed retrieving service from state store: %v", err)
+			}
+
+			svc := svcRaw.(*structs.ServiceNode)
+			if err := tx.Insert("index", &IndexEntry{serviceKindIndexName(svc.ServiceKind), idx}); err != nil {
+				return fmt.Errorf("failed updating index: %s", err)
+			}
 		} else {
```

Comment:
What about this else case? If the check is node level it could still cause a service of a particular kind to change status.

Do we not need to also set this per-kind index in `updateAllServiceIndexesOfNode`?

Reply:
I will have to double check but I thought that it already did end up doing just that.

Reply:
I double checked and that is already implemented.